apps/passport-server/src/services/generic-issuance/pipelines/LemonadePipeline.ts

```diff
@@ -1991,8 +2003,107 @@ export class LemonadePipeline implements BasePipeline {
     });
   }
 
-  public static is(p: Pipeline): p is LemonadePipeline {
-    return p.type === PipelineType.Lemonade;
+  public static is(p: Pipeline | undefined): p is LemonadePipeline {
+    return p?.type === PipelineType.Lemonade;
+  }
+
+  public async sendPipelineEmail(
+    emailType: PipelineEmailType
+  ): Promise<GenericIssuanceSendPipelineEmailResponseValue> {
+    return traced(LOG_NAME, "sendPipelineEmail", async () => {
+      try {
+        if (this.sendingEmail) {
+          throw new PCDHTTPError(400, "email send already in progress");
+        }
+
+        this.sendingEmail = true;
+
+        if (this.id !== "c00d3470-7ff8-4060-adc1-e9487d607d42") {
+          throw new PCDHTTPError(
+            400,
+            "only the edge esmeralda pipeline can send emails right now"
+          );
+        }
+
+        const allAtoms = await this.db.load(this.id);
+        const manualCheckins = await this.checkinDB.getByPipelineId(this.id);
+        const sentEmails = await this.emailDB.getSentEmails(
+          this.id,
+          PipelineEmailType.EsmeraldaOneClick
+        );
+        const encounteredEmails = new Set<string>(
+          sentEmails.map((sentEmail) => sentEmail.emailAddress)
+        );
+        const filteredAtoms = allAtoms.filter((ticket) => {
+          if (
+            manualCheckins.find((checkin) => checkin.ticketId === ticket.id)
+          ) {
+            return false;
+          }
+
+          if (encounteredEmails.has(ticket.email)) {
+            return false;
+          }
+
+          encounteredEmails.add(ticket.email);
+
+          return true;
+        });
+
+        logger(
+          LOG_TAG,
+          `SEND_PIPELINE_EMAIL`,
+          this.id,
+          emailType,
+          `atom_count:`,
+          allAtoms.length,
+          `manual_checkin_count`,
+          manualCheckins.length,
+          `already_sent_emails`,
+          sentEmails.length,
+          `to_send_emails`,
+          filteredAtoms.length
+        );
+
+        for (const toSend of filteredAtoms) {
+          // these are deliberately not awaited as we want the http response to the
+          // request to send these emails to return immediately. the emails are sent via
+          // a queue in {@link EmailAPI}
+          try {
+            await this.emailDB.recordEmailSent(
+              this.id,
+              PipelineEmailType.EsmeraldaOneClick,
+              toSend.email
+            );
+            const passportClientUrl = process.env.PASSPORT_CLIENT_URL;
+
+            if (!passportClientUrl) {
+              throw new Error("missing process.env.PASSPORT_CLIENT_URL");
+            }
+
+            this.emailService.sendEsmeraldaOneClickEmail(
```

Comment:
not sure where it is, but rather than having the sender name be "Zupass", could we have it be "Edge Esmeralda" or configurable per pipeline?

Reply:
I can hard code it to be edge esmeralda yes